tests/E2E/test/campaigns/common_scenarios/linkwidget.js

```diff
@@ -7,29 +7,65 @@ module.exports = {
     scenario('Create new "Link Widget" block ', client => {
       test('should go to "Design - Link Widget" page', () => client.goToSubtabMenuPage(Menu.Improve.Design.design_menu, Menu.Improve.Design.link_widget_submenu));
       test('should click on "New block" button', () => client.waitForExistAndClick(LinkWidget.new_block_button));
-      test('should set the "Name of the link block"', () => client.waitAndSetValue(LinkWidget.name_of_the_link_block_input, name + " " + date_time));
+      test('should set the "Name of the link block in english"', () => client.waitAndSetValue(LinkWidget.name_of_the_link_block_input.replace('%lang','1'), name + " " + date_time));
+      test('should switch input language to french', () => {
+        return promise
+          .then(() => client.waitForExistAndClick(LinkWidget.name_of_the_link_block_lang_button))
+          .then(() => client.waitForVisibleAndClick(LinkWidget.name_of_the_link_block_lang_span.replace('%lang','fr')));
+      });
+      test('should set the "Name of the link block in french', () => client.waitAndSetValue(LinkWidget.name_of_the_link_block_input.replace('%lang','2'), name + " " + date_time));
       test('should choose the hook "displayFooter"', () => client.waitAndSelectByVisibleText(LinkWidget.hook_select, hook));
-      test('should select All the "content pages"', () => client.waitForExistAndClick(LinkWidget.select_all_content_page));
-      test('should select All the "product pages"', () => client.waitForExistAndClick(LinkWidget.select_all_product_page));
-      test('should select All the "static content"', () => client.waitForExistAndClick(LinkWidget.select_all_static_content));
-      test('should set "Name" input', () => client.waitAndSetValue(LinkWidget.first_custom_content_name_input, 'Custom Name ' + date_time));
-      test('should set "URL" input', () => client.waitAndSetValue(LinkWidget.first_custom_content_url_input, 'Custom Url ' + date_time));
-      test('should click on "Add" button', () => client.waitForExistAndClick(LinkWidget.add_custom_content_button));
+      test('should select All the "content pages"', async () => {
+        let i = 1;
+        do{
+          await client.isVisible(LinkWidget.select_all_content_page.replace('%POS',i));
+          if(global.isVisible)
+            client.scrollWaitForExistAndClick(LinkWidget.select_all_content_page.replace('%POS',i));
+          i++;
+        }while(global.isVisible);
```

Comment:
Maybe I'm paranoid but ... no possible infinite loop ? 😛

Reply:
Since select_all_content_page is a list and global.isVisible is update each time in client.isVisible 
than no, it's not a infinite loop